### PR TITLE
CSHARP-4660: Add search index management helpers

### PR DESCRIPTION
**Pull request by BorisDog**
Tested manually.
E2E testing will be done in [DRIVERS-2630](https://jira.mongodb.org/browse/DRIVERS-2630)

### CODE REVIEW

File: src/MongoDB.Driver/Search/IMongoSearchIndexManager.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        /// </summary>
+        /// <param name="name">The index name.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A task.</returns>
+        Task DropOneAsync(string name, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Lists the search indexes.
+        /// </summary>
+        /// <param name="name ">Name of the index.</param>
```
Comment (sanych-sun) [inline src/MongoDB.Driver/Search/IMongoSearchIndexManager.cs:108]: Extra space

File: src/MongoDB.Driver/Search/IMongoSearchIndexManager.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        /// A cursor.
+        /// </returns>
+        IAsyncCursor<BsonDocument> List(string name = null, AggregateOptions aggregateOptions = null, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Lists the search indexes.
+        /// </summary>
+        /// <param name="name ">Name of the index.</param>
```
Comment (sanych-sun) [inline src/MongoDB.Driver/Search/IMongoSearchIndexManager.cs:119]: Extra space

File: tests/MongoDB.Driver.Tests/Specifications/index-management/IndexManagementUnifiedTestRunner.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        {
+            // protected properties
+            protected override string PathPrefix => "MongoDB.Driver.Tests.Specifications.index_management.tests.";
+
+            // protected methods
+            protected override IEnumerable<JsonDrivenTestCase> CreateTestCases(BsonDocument document)
+            {
+                foreach (var testCase in base.CreateTestCases(document))
+                {
+                    foreach (var async in new[] { false, true })
```
Comment (sanych-sun) [inline tests/MongoDB.Driver.Tests/Specifications/index-management/IndexManagementUnifiedTestRunner.cs:54]: I do remember we unified such code. Can we reuse that implementation? or search index management tests are not in unified format?

**Review by sanych-sun — APPROVED** [review]
LGTM with minor suggestions

**Review by JamesKovacs — APPROVED** [review]
LGTM

File: src/MongoDB.Driver/Search/IMongoSearchIndexManager.cs (hunk 1 above)
Comment (BorisDog) [reply src/MongoDB.Driver/Search/IMongoSearchIndexManager.cs]: Thanks.

File: src/MongoDB.Driver/Search/IMongoSearchIndexManager.cs (hunk 2 above)
Comment (BorisDog) [reply src/MongoDB.Driver/Search/IMongoSearchIndexManager.cs]: Fixed.

File: tests/MongoDB.Driver.Tests/Specifications/index-management/IndexManagementUnifiedTestRunner.cs (hunk 3 above)
Comment (BorisDog) [reply tests/MongoDB.Driver.Tests/Specifications/index-management/IndexManagementUnifiedTestRunner.cs:54]: Yes we did not merge that PR yes.

**Pull request merged**